optimize: undo_log table check optimize

laywin opened the pull request:
<!-- Please make sure you have read and understood the contributing guidelines -->

- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did
https://github.com/seata/seata/issues/6022

### Ⅱ. Does this pull request fix one issue?
<!-- If that, add "fixes #xxx" below in the next line, for example, fixes #97. -->
fixes #6022

### Ⅲ. Why don't you add test cases (unit test/integration test)? 


### Ⅳ. Describe how to verify it


### Ⅴ. Special notes for reviews

laywin committed:
undo_log table check optimize

laywin committed:
undo_log table check optimize

laywin committed:
optimize & add modify record

laywin committed:
optimize & add modify record

laywin committed:
adjust indentation

laywin committed:
adjust indentation

funky-eyes commented on rm-datasource/src/test/java/io/seata/rm/RMHandlerATTest.java:
Write a test case for the newly modified code.

laywin replied:
done

laywin committed:
add unit test

laywin committed:
move pr record

laywin committed:
Merge remote-tracking branch 'origin/2.x' into 2.x

# Conflicts:
#	changes/en-us/2.0.0.md
#	changes/zh-cn/2.0.0.md
#	rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java
#	rm-datasource/src/test/java/io/seata/rm/RMHandlerATTest.java

laywin committed:
modify unit test

laywin committed:
add Github ID

funky-eyes commented on changes/zh-cn/2.x.md:
```diff
@@ -9,7 +9,7 @@
 
 ### optimize:
-- [[#PR_NO](https://github.com/seata/seata/pull/PR_NO)] 准确简要的PR描述
+- [[#6031](https://github.com/seata/seata/pull/6031)] 添加undo_log表的存在性校验
```
该版本下首次提交pr，作者信息需要进行登记一下
For the first PR submission in this version, the author's information needs to be registered.

laywin replied:
是这个吗，有登记额
Is this it? It's registered
<img width="1322" alt="image" src="https://github.com/seata/seata/assets/22276474/6092c2c6-91e8-4133-bc01-ce30fc4b0c3c">

funky-eyes replied:
你看下中文的那份文件是否没登记
Check if the Chinese file is not registered.

laywin replied:
done

funky-eyes commented on rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:
```diff
@@ -143,6 +153,32 @@ private boolean isPolardbXProduct() {
+            try {
+                undoLogManager = UndoLogManagerFactory.getUndoLogManager(dbType);
+            } catch (EnhancedServiceNotFoundException e) {
+                LOGGER.error("can't find undoLogManager service provider for dbtype: {}", dbType);
```
我建议这里直接抛异常出去，说明用户再使用一个我们不支持的数据库，要提示这个数据库不支持
I suggest throwing an exception directly here, indicating that if the user uses a database that we do not support, we need to prompt that the database is not supported.

laywin replied:
没有抛出去，是因为很多单测使用了数据库代理，会导致单测失败。
It is not thrown out because many unit tests use database proxy, which can be lead it fail.

funky-eyes replied:
> 没有抛出去，是因为很多单测使用了数据库代理，会导致单测失败。 It is not thrown out because many unit tests use database proxy, which can be lead it fail.

有没有什么办法在单测的时候不进行校验undolog table？或者增加一个开关，这个开关允许用户配置check，如果用户关闭，就不进行check，然后我们默认开启，在单测的时候关闭check
Is there any way to not check undolog table during single test or add a switch that allows user to configure check and if user turns it off, no check is performed, then we turn it on by default and turn off check during single test

laywin replied:
update mockito version to 4.5.1 same as seata server's mockito dependency to support static mock. when unit test. mock undolog table check.

funky-eyes committed:
Merge branch '2.x' into 2.x

laywin committed:
1. update mockito version to 4.5.1 same as seata server to support static mock
2. throw exception when the db type not support in AT mode
3. add author register info

laywin committed:
Merge remote-tracking branch 'origin/2.x' into 2.x

# Conflicts:
#	changes/zh-cn/2.x.md

laywin committed:
import optimize

funky-eyes reviewed (approved):
LGTM
